spec/unit/driverProviders/direct_test.js

```diff
@@ -57,9 +64,43 @@ describe('direct connect', function() {
       };
       var errorFound = false;
       try {
-        webdriver = require('../../../built/driverProviders/direct')(config);
+        webdriver = new Direct(config);
+        webdriver.getNewDriver();
+      } catch(e) {
+        errorFound = true;
+        expect(e.code).toBe(BrowserError.CODE);
+      }
+      expect(errorFound).toBe(true);
+    });
+  });
+
+  describe('binary does not exist', () => {
+    beforeEach((done) => {
+      // webdriver-manager clean
+      var argv = {
+        '_': ['update'],
+        'standalone': false,
+        'gecko': false
+      };
+      webdriverUpdate.run(JSON.parse(JSON.stringify(argv))).then(() => {
+        const SeleniumConfig = require('webdriver-manager/built/lib/config').Config;
+        rimraf.sync(path.resolve(SeleniumConfig.getSeleniumDir(), 'update-config.json'));
```

Comment:
I don't like this because:
1) You're actually changing the file system, which is an unexpected side effect of running these tests
2) Unit tests shouldn't require you to have the filesystem set up in any particular  setup

Please either use a mock file system of some kind or change `lib/driverProviders/local.ts` so you can call `addDefaultBinaryLocs_()` without it interacting with the file system.